server/neptune/workflows/internal/terraform/state/store.go

```diff
@@ -85,6 +87,18 @@ func (s *WorkflowStore) UpdatePlanJobWithStatus(status JobStatus, options ...Upd
 }
 
 func (s *WorkflowStore) UpdateApplyJobWithStatus(status JobStatus, options ...UpdateOptions) error {
+	var timeStamp time.Time
+	for _, o := range options {
+		timeStamp = o.Timestamp
+	}
+
+	// add start and end time for apply job
+	if status == InProgressJobStatus {
+		s.state.Apply.StartTime = timeStamp
+	} else if status == FailedJobStatus || status == SuccessJobStatus {
+		s.state.Apply.EndTime = timeStamp
+	}
```

Comment:
Does this make sense to the caller? As a caller I pass in an arbitrary timestamp with my status not knowing why or what it does unless I look at this implementation?

I think it makes more sense to just pass in StartTime and EndTime separately as part of the options.